apps/backend/genesis_test-case/src/grpc-server.ts

```diff
@@ -0,0 +1,440 @@
+import * as grpc from '@grpc/grpc-js';
+import * as protoLoader from '@grpc/proto-loader';
+import * as path from 'path';
+import { 
+  getGenres, 
+  getTracks, 
+  getTrackBySlug, 
+  createTrack, 
+  updateTrack, 
+  deleteTrack, 
+  deleteMultipleTracks,
+  saveAudioFile,
+  deleteAudioFile,
+  getTrackById
+} from './utils/db';
+import { createSlug } from './utils/slug';
+import { updateTracksList, startWSServer } from './ws-server';
+import { 
+  IGenreService, 
+  ITrackService,
+  GetAllGenresResponse,
+  GetAllTracksResponse,
+  GetTrackBySlugResponse,
+  CreateTrackResponse,
+  UpdateTrackResponse,
+  DeleteTrackResponse,
+  DeleteMultipleTracksResponse,
+  UploadTrackFileResponse,
+  DeleteTrackFileResponse
+} from './types/grpc';
+
+const PROTO_PATH = path.resolve(__dirname, 'music_service.proto');
+
+const packageDefinition = protoLoader.loadSync(PROTO_PATH, {
+  keepCase: true,
+  longs: String,
+  enums: String,
+  defaults: true,
+  oneofs: true
+});
+
+const musicProto = grpc.loadPackageDefinition(packageDefinition) as any;
```

Comment:
Just a suggestion, you could generate typescript types from `.proto` files with buf. Since you already have everything written, you can just ignore this, or implement if you want. This would also make implementing endpoints easier (you can see lower in the docs).
https://connectrpc.com/docs/node/getting-started#generate-code